[dist_optim] add warning to distributed optimizer

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#50630 [dist_optim] add warning to distributed optimizer**

Add a warning log to distributed optimizer, to warn user the optimizer
is created without TorchScript support.

Differential Revision: [D25932777](https://our.internmc.facebook.com/intern/diff/D25932777)

wanchaol committed:
[dist_optim] add warning to distributed optimizer

Add a warning log to distributed optimizer, to warn user the optimizer
is created without TorchScript support.

[ghstack-poisoned]

wanchaol committed:
Update on "[dist_optim] add warning to distributed optimizer"


Add a warning log to distributed optimizer, to warn user the optimizer
is created without TorchScript support.

[ghstack-poisoned]

rohan-varma commented on torch/distributed/optim/optimizer.py:
```diff
@@ -204,6 +206,13 @@ def __init__(self, optimizer_class, params_rref, *args, **kwargs):
+                "slow computation time in multithreading environment (i.e. Distributed Model "
+                "Parallel training on CPU) due to the python Global Interpreter Lock (GIL). "
+                "Please file an issue if you need this optimizer in TorchScript. ",
+                str(optimizer_class)
```
Nit: use f-string

rohan-varma commented on torch/distributed/optim/optimizer.py:
```diff
@@ -204,6 +206,13 @@ def __init__(self, optimizer_class, params_rref, *args, **kwargs):
             optimizer_new_func = _new_script_local_optimizer
         else:
+            logger.warn(
+                "Creating the optimizer %s without TorchScript support, this might results in "
```
"this might results" -> "this might result"

wanchaol committed:
Update on "[dist_optim] add warning to distributed optimizer"


Add a warning log to distributed optimizer, to warn user the optimizer
is created without TorchScript support.

Differential Revision: [D25932777](https://our.internmc.facebook.com/intern/diff/D25932777)

[ghstack-poisoned]

wanchaol committed:
Update on "[dist_optim] add warning to distributed optimizer"


Add a warning log to distributed optimizer, to warn user the optimizer
is created without TorchScript support.

Differential Revision: [D25932777](https://our.internmc.facebook.com/intern/diff/D25932777)

[ghstack-poisoned]

wanchaol committed:
Update on "[dist_optim] add warning to distributed optimizer"


Add a warning log to distributed optimizer, to warn user the optimizer
is created without TorchScript support.

Differential Revision: [D25932777](https://our.internmc.facebook.com/intern/diff/D25932777)

[ghstack-poisoned]

wanchaol committed:
Update on "[dist_optim] add warning to distributed optimizer"


Add a warning log to distributed optimizer, to warn user the optimizer
is created without TorchScript support.

Differential Revision: [D25932777](https://our.internmc.facebook.com/intern/diff/D25932777)

[ghstack-poisoned]

wanchaol committed:
Update on "[dist_optim] add warning to distributed optimizer"


Add a warning log to distributed optimizer, to warn user the optimizer
is created without TorchScript support.

Differential Revision: [D25932777](https://our.internmc.facebook.com/intern/diff/D25932777)

[ghstack-poisoned]

wanchaol committed:
Update on "[dist_optim] add warning to distributed optimizer"


Add a warning log to distributed optimizer, to warn user the optimizer
is created without TorchScript support.

Differential Revision: [D25932777](https://our.internmc.facebook.com/intern/diff/D25932777)

[ghstack-poisoned]
